crates/bin/node/src/response_cache.rs

```diff
@@ -0,0 +1,72 @@
+use std::{
+    fmt::Debug,
+    time::{Duration, Instant},
+};
+
+use crate::errors;
+use dashmap::DashMap;
+use node::{MeltResponse, MintQuoteResponse, MintResponse, SwapResponse};
+
+pub trait ResponseCache<K, V> {
+    // Basic operations
+    fn get(&self, key: &K) -> Option<V>;
+    fn insert(&self, key: K, value: V) -> Result<(), errors::Error>;
+    fn remove(&self, key: &K) -> bool;
+
+    // For persistence when shutting down
+    fn persist(&self) -> Result<(), errors::Error> {
+        Ok(())
```

Comment:
No need to implement it in this PR if you don't feel like doing it, just add a comment:
`# TODO: `